google/cloud/pubsublite/internal/wire/partition_count_watching_publisher.py

```diff
@@ -56,8 +56,9 @@ async def __aenter__(self):
         return self
 
     async def __aexit__(self, exc_type, exc_val, exc_tb):
-        self._partition_count_poller.cancel()
-        await wait_ignore_cancelled(self._partition_count_poller)
+        if hasattr(self, '_partition_count_poller'):
```

Comment:
hasattr is a code smell. Instead, change the type of _partition_count_poller to Optional[asyncio.Future], set it to None in the __init__ method, and test `if self._partition_count_poller is not None:`

Reply:
Done.